app/controllers/concerns/pagy_pagination.rb

```diff
@@ -0,0 +1,63 @@
+module PagyPagination
+  extend ActiveSupport::Concern
+
+  def jsonapi_pagination(collection)
+    collection.present? && pagy_scope.present? ? pagination_links : {}
+  end
+
+private
+
+  # child must define pagy_scope method for paginations related items
+  def pagy_scope; end
```

Comment:
Still not sure this is the clearest way to do it but if we are going to use this then we should raise something here if the including class is required to implement this. Otherwise it will fail silently. I always use `NotImplementedError` but that's not really what it's supposed to be for I don't think. Sounds right though :)

Reply:
``` ruby
def pagy_scope; 
  raise NotImplementedError, "must define pagy_scope method"
end
```

you mean do something like this?

Reply:
Yes. Well, yes if we're going to take this approach. I still think having an explicit `paginate(scope)` method is clearer in the controllers.

Reply:
opted with `paginate(scope)`